### DISCUSSION TITLE
Short URL doesn't work on Synology Docker

### DISCUSSION THREAD

**Discussion by MuadDibIII** [shlinkio/shlink#1602] [Help wanted]
Hello,

I just install Shlink on my Synology NAS using the MariaDB package from Synology.
I also installed the Shlink web app container, and it connects correctly to the Shink server using the generated API key.

After generating a short URL, it works, but when I try reach the target site, I getting a 404 error.

Here are some technical details:

- NAS URL: **https://sub.doma.in:XXXX** (my NAS uses a subdomain and a custom port)
- Shlink container: **sub.doma.in:YYYY** (same NAS URL, another port)
- Shlink client: **http://sub.doma.in:ZZZZ**
- DEFAULT_DOMAN=**sub.doma.in:XXXX**
- IS_HTTPS_ENABLED=**true**

My short link is something like **https://sub.doma.in:XXXX/J9U9u**

Is there any missing configuration? I think that I have somehow to point the **/J9U9u** to Shlink server, so it can correctly forward the request, but how?

Thanks in advance for your help,

Best regards,
Marc

**Comment by acelaya** [comment]
I'm a bit confused about why there are three different ports in the equation. Whatever you set in `DEFAULT_DOMAN` should match your public URL, so I guess it should be https://sub.doma.in:XXXX/

**Reply by MuadDibIII** [reply]
Hello acelaya,

Thanks for your reply. I think I was not very clear on my ports. Here is a more detailed description:

- XXXX: public port of the NAS, the one that I can connect to DSM
- YYYY: is the Shlink port, the one that is mapped to the port 8080 on the Docker compose file
- ZZZZ: is the port that I use to access the Shlink web client container, that is mapped to the port 80 on it's container.

Now as I declared on my DEFAULT_DOMAIN variable, my short URLs have my NAS public port (XXXX). When I try to use them, I have a 404 error, probably because the NAS doesn't now how to handle the "/abcd" apend.

What do you think? Do you see anything wrong in my install?

Thanks again,

Best regards,
Marc

**Comment by MuadDibIII** [comment]
The question that I'm asking myself is: how Shlink "takes control" of the query? When the NAS receives the short link, how can it transfer to Shlink so it can correctly redirect it to the desired target?

**Reply by acelaya** [reply]
I'm afraid I don't really know how docker works on synology, or how does it serve HTTP requests and exposes services 😅

**Reply by MuadDibIII** [reply]
Hello, 

Sorry for the delay on answering. I don't know why, but everything just stop working, I couldn't access the server anymore from the web client.

At the end I found YOURLS, it has less features but at least it's more straight forward to install from the Synology NAS.

Thanks again,
Marc
